monai/apps/auto3d/data_analyzer.py

```diff
@@ -0,0 +1,618 @@
+
+        """
+        self.data["image"] = batch_data["image"].to(self.device)
+        self.data["label"] = batch_data["label"].to(self.device)
+        self.data["image_meta_dict"] = batch_data["image_meta_dict"]
+        self.data["label_meta_dict"] = batch_data["label_meta_dict"]
+        case_stats = {}
+        for func in self.functions:
+            case_stats.update(func())
+        return case_stats
+
+    def _get_case_summary(self):
+        """
+        Function to combine case stats. The stats for each case is stored in self.results['stats_by_cases'].
+        Each case stats is a dictionary. The function first get all the leaf-keys of self.gather_summary.
+        self.gather_summary is a dictionary of the same structure with the final summary yaml
+        output (self.results['stats_summary']), because it is updated by case_stats_summary.
+        The operations is retrived by recursive_getvalue and the combined value is calculated.
+
+        summarize the results from each case using functions _intensity_summary, _stats_opt_summary.
+        """
+        # initialize gather_summary
+        [func() for func in self.functions_summary]
+        # read
+        key_chains = recursive_getkey(self.gather_summary)
+        self.results["stats_summary"] = copy.deepcopy(self.gather_summary)
+        for key_chain in key_chains:
+            opt = recursive_getvalue(self.gather_summary, key_chain)
+            value = opt(
+                list(filter(None, [recursive_getvalue(case, key_chain) for case in self.results["stats_by_cases"]]))
+            )
+            recursive_setvalue(key_chain, value, self.results["stats_summary"])
+
+    def get_all_case_stats(self) -> Dict:
+        """
+        Get all case stats. Caller of the DataAnalyser class. The function iterates datalist and
+        call get_case_stats to generate stats. Then get_case_summary is called to combine results.
+
+        Returns
+            - the data statistics dictionary
+        """
+        start = time.time()
+        self.results["stats_summary"] = {}
+        self.results["stats_by_cases"] = []
+        s = start
+        if not has_tqdm:
+            warnings.warn("tqdm is not installed. not displaying the caching progress.")
+
+        for batch_data in tqdm(self.dataset) if has_tqdm else self.dataset:
+            images_file = batch_data[0]["image_meta_dict"]["filename_or_obj"]
+            label_file = batch_data[0]["label_meta_dict"]["filename_or_obj"]
+            logger.debug(f"Load data spent {time.time() - s}")
+            case_stat = {"image": images_file, "label": label_file}
+            case_stat.update(self.get_case_stats(batch_data[0]))
+            self.results["stats_by_cases"].append(case_stat)
+            logger.debug(f"Process data spent {time.time() - s}")
+            s = time.time()
+        self._get_case_summary()
+        ConfigParser.export_config_file(self.results, self.output_path, fmt="yaml")
```

Comment:
I changed this line so that the yaml output is more compact, would be great to make this option configurable
```suggestion
        ConfigParser.export_config_file(self.results, self.output_path, fmt="yaml", default_flow_style=None)
```

```yaml
    label_60:
      image_intensity:
      - {max: 104.0, mean: 77.4637451171875, median: 81.0, min: 24.0, percentile_00_5: 33.61499786376953,
        percentile_10_0: 58.0, percentile_90_0: 90.0, percentile_99_5: 99.385009765625,
        stdev: 13.304381370544434}
      ncomponents: 1
      shape:
      - [28, 17, 21]
    label_62:
      image_intensity:
      - {max: 65.0, mean: 57.79999923706055, median: 57.0, min: 51.0, percentile_00_5: 51.13999938964844,
        percentile_10_0: 53.400001525878906, percentile_90_0: 62.0, percentile_99_5: 64.79000091552734,
        stdev: 3.618471145629883}
      ncomponents: 1
      shape:
      - [4, 2, 4]
    label_7:
      image_intensity:
      - {max: 118.0, mean: 87.16324615478516, median: 88.0, min: 24.0, percentile_00_5: 51.96999740600586,
        percentile_10_0: 78.0, percentile_90_0: 96.0, percentile_99_5: 104.0, stdev: 7.87377405166626}
      ncomponents: 5
      shape:
      - [51, 37, 36]
      - [5, 1, 3]
      - [5, 1, 4]
      - [4, 1, 4]
      - [3, 1, 2]
    label_8:
      image_intensity:
      - {max: 93.0, mean: 63.62257766723633, median: 66.0, min: 12.0, percentile_00_5: 28.0,
        percentile_10_0: 48.0, percentile_90_0: 75.0, percentile_99_5: 83.0, stdev: 10.754353523254395}
      ncomponents: 1
      shape:
      - [64, 41, 65]
    labels: [0, 2, 3, 4, 5, 7, 8, 10, 11, 12, 13, 14, 15, 16, 17, 18, 24, 26, 28,
      41, 42, 43, 44, 46, 47, 49, 50, 51, 52, 53, 54, 58, 60, 62]
    pixel_percentage: {0: 0.8774868249893188, 2: 0.01772916316986084, 3: 0.033243417739868164,
      4: 0.00032341480255126953, 5: 1.7404556274414062e-05, 7: 0.0011676549911499023,
      8: 0.0053931474685668945, 10: 0.0007014274597167969, 11: 0.00029075145721435547,
      12: 0.0004571676254272461, 13: 0.00014960765838623047, 14: 6.365776062011719e-05,
      15: 0.00012159347534179688, 16: 0.0017879009246826172, 17: 0.00032699108123779297,
      18: 0.0001634359359741211, 24: 6.008148193359375e-05, 26: 5.137920379638672e-05,
      28: 0.00035822391510009766, 41: 0.016874313354492188, 42: 0.033881306648254395,
      43: 0.00035190582275390625, 44: 1.8358230590820312e-05, 46: 0.0010356903076171875,
      47: 0.005444884300231934, 49: 0.0007225275039672852, 50: 0.0002931356430053711,
      51: 0.00048029422760009766, 52: 0.0001392364501953125, 53: 0.00029754638671875,
      54: 0.00016987323760986328, 58: 4.7326087951660156e-05, 60: 0.0003485679626464844,
      62: 1.7881393432617188e-06}
```